### PR TITLE
Create an analytical storage middleware layer

### LINKED ISSUE #12728: Convert the current PointLogWriter to be a TaskService middleware
The current implementation of the Log writer is built as a stand alone log/run Writer. To simplify the way we do these function calls we can make a middleware layer in top of the kv task service that writes to the analytical store when the run in finalized. This will allow writer to only write one time and pull the data in a simple and precise way.

The middleware here will also act as a log/run reader for data that isn't in the kv service.

### PR DESCRIPTION
**Pull request by lyondhill**
Closes #12728

### CODE REVIEW

File: task.go [hunk 1]
```diff
@@ -71,6 +71,11 @@ func (r *Run) ScheduledForTime() (time.Time, error) {
 	return time.Parse(time.RFC3339, r.ScheduledFor)
 }
 
+// StartedAtTime gives the time.Time that the run is scheduled for.
```
Comment (mark-rushakoff) [inline task.go]: ... the time that the run started?

File: ui/package-lock.json [hunk 2]
```diff
@@ -6176,21 +6176,20 @@
       "dependencies": {
         "abbrev": {
           "version": "1.1.1",
-          "resolved": false,
+          "resolved": "",
```
Comment (mark-rushakoff) [inline ui/package-lock.json]: Can you undo your changes to this file?

File: task/servicetest/servicetest.go [hunk 3]
```diff
@@ -1188,11 +1197,101 @@ func testRetryAcrossStorage(t *testing.T, sys *System) {
...
+		t.Fatal(err)
+	}
+
+	logs, _, err := sys.TaskService.FindLogs(sys.Ctx, influxdb.LogFilter{Task: task.ID})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(logs) != 7 {
+		for _, log := range logs {
+			fmt.Printf("log: %+v\n", log)
```
Comment (mark-rushakoff) [inline task/servicetest/servicetest.go]: This should be `t.Logf` so it displays correctly with the `t.Fatalf` below.

File: task/servicetest/servicetest.go [hunk 4]
```diff
@@ -1188,11 +1197,101 @@ func testRetryAcrossStorage(t *testing.T, sys *System) {
...
+	if err := sys.TaskControlService.UpdateRunState(sys.Ctx, task.ID, rc1.Created.RunID, startedAt, backend.RunStarted); err != nil {
+		t.Fatal(err)
+	}
+
+	// Mark the second run finished.
+	if err := sys.TaskControlService.UpdateRunState(sys.Ctx, task.ID, rc1.Created.RunID, startedAt.Add(time.Second), backend.RunSuccess); err != nil {
+		t.Fatal(err)
+	}
+
+	sys.TaskControlService.AddRunLog(sys.Ctx, task.ID, rc0.Created.RunID, time.Now(), "0-0")
```
Comment (mark-rushakoff) [inline task/servicetest/servicetest.go]: Can you add more comments explaining the intent in this new test?

I think something is going on between rc0 and rc1 but it's not really obvious to me.

File: task/backend/analytical_storage_test.go [hunk 5]
```diff
@@ -0,0 +1,132 @@
+package backend_test
+
+import (
+	"context"
+	// "fmt"
```
Comment (mark-rushakoff) [inline task/backend/analytical_storage_test.go]: Remove comment.

File: task/backend/analytical_storage_test.go [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+	if err := os.RemoveAll(lrw.rootDir); err != nil {
+		t.Error(err)
+	}
+}
+
+func newAnalyticalBackend(t *testing.T, orgSvc influxdb.OrganizationService, bucketSvc influxdb.BucketService) *analyticalBackend {
+	// Mostly copied out of cmd/influxd/main.go.
+	logger := zaptest.NewLogger(t)
+
+	rootDir, err := ioutil.TempDir("", "task-logreaderwriter-")
```
Comment (mark-rushakoff) [inline task/backend/analytical_storage_test.go]: Does rootDir get removed somewhere?

File: ui/package-lock.json (hunk 2 above)
Comment (lyondhill) [reply ui/package-lock.json]: yah. my bad :(

File: task/backend/analytical_storage_test.go (hunk 6 above)
Comment (lyondhill) [reply task/backend/analytical_storage_test.go]: in the close method I remove it. But it is currently possible for it to not be removed if we fail to setup the engine. I will fix that.

**Pull request merged**